Makefile

```diff
@@ -31,6 +31,19 @@ DIST = $(DISTDIR)/.done
 GIT_HEAD_HASH = $(shell git rev-parse HEAD)
 BUILDBRANCH ?= staging
 
+# "Baked in" environment variables
+# for built JS bundle
+NODE_ENV = production
+GITHUB_API_ENDPOINT = https://api.github.com
+STORE_DEVPORTAL_URL = https://myapps.developer.ubuntu.com/dev
+
+target staging:
+	BASE_URL = https://build.staging.snapcraft.io
+
+target production:
+	BASE_URL = https://build.snapcraft.io
+
```

Comment:
This is quite unfortunate; it means that we won't be using exactly the same bundle that we tested on staging when we promote it to production.  Is there any way to avoid this while preserving the other benefits you're trying to gain?  I'd really strongly prefer the bundles to be identical, rather than testing one build and then hoping that nothing mysterious went wrong with the other similar build before promoting it to production; in other words, I view separate staging and production builds as an antifeature.

As for syntax, this doesn't look right; the variables would need to be exported, and the syntax for setting a target-specific variable looks more like `staging: export BASE_URL = https://build.staging.snapcraft.io`.  But of course this is all contingent on the above comment.